assemblies/assembly-planner-java.adoc

```diff
@@ -0,0 +1,750 @@
++
+----
+...
+INFO  |            | Room A     | Room B     | Room C     |
+INFO  |------------|------------|------------|------------|
+INFO  | MON 08:30  | English    | Math       |            |
+INFO  |            | I. Jones   | A. Turing  |            |
+INFO  |            | 9th grade  | 10th grade |            |
+INFO  |------------|------------|------------|------------|
+INFO  | MON 09:30  | History    | Physics    |            |
+INFO  |            | I. Jones   | M. Curie   |            |
+INFO  |            | 9th grade  | 10th grade |            |
+...
+----
+
+. Verify the console output. Does it conform to all hard constraints?
+What happens if you comment out the `roomConflict` constraint in `TimeTableConstraintProvider`?
+
+The `info` log shows what OptaPlanner did in those five seconds:
+
+[source,options="nowrap"]
+----
+... Solving started: time spent (33), best score (-8init/0hard/0soft), environment mode (REPRODUCIBLE), random (JDK with seed 0).
+... Construction Heuristic phase (0) ended: time spent (73), best score (0hard/0soft), score calculation speed (459/sec), step total (4).
+... Local Search phase (1) ended: time spent (5000), best score (0hard/0soft), score calculation speed (28949/sec), step total (28398).
+... Solving ended: time spent (5000), best score (0hard/0soft), score calculation speed (28524/sec), phase total (2), environment mode (REPRODUCIBLE).
+----
+
+== Test the application
+A good application includes test coverage. Test the constraints and the solver in your timetable project.
+
+include::{enterprise-dir}/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/optimizer/business-optimizer-test-solver-quarkus-proc.adoc[leveloffset=+2]
+include::{enterprise-dir}/optimizer/business-optimizer-logging-quarkus-proc.adoc[leveloffset=+1]
+
+== Using Micrometer and Prometheus to monitor your school timetable OptaPlanner Java application
+
+OptaPlanner exposes metrics through https://micrometer.io/[Micrometer], a metrics instrumentation library for Java applications. You can use Micrometer with Prometheus to monitor the OptaPlanner solver in the school timetable application.
+
+To configuring a plain Java OptaPlanner application to use Micrometer, you must
+
+
+.Prerequisites
+* You have created the OptaPlanner school timetable application with Java.
+
+.Procedure
+. Add the Micrometer Promtheus dependencies to the school timetable `pom.xml` file:
++
+[source]
+----
+<dependency>
+ <groupId>io.micrometer</groupId>
+ <artifactId>micrometer-registry-prometheus</artifactId>
+ <version>1.7.3</version>
+</dependency>
+<dependency>
+ <groupId>io.micrometer</groupId>
+ <artifactId>micrometer-core</artifactId>
+ <version>1.7.3</version>
+</dependency>
+----
+
+. Add the following lines to the top of the `pom.xml` file so Prometheus can scrap data from `com.sun.net.httpserver.HttpServer`  before the solutions starts:
```

Comment:
This shoud be added not in top pf pom xml, this shoudl be added to the TimeTableApp.java
to the top of the main method 
+ relative imports 

```
import io.micrometer.core.instrument.Metrics;
import io.micrometer.prometheus.PrometheusConfig;
import io.micrometer.prometheus.PrometheusMeterRegistry;
```

The whole file  TimeTableApp.java should be like this one https://github.com/dupliaka/hello-world-micrometer/blob/191734c7c75e70682a6c2d39e79ffd5893cbdee3/src/main/java/org/acme/schooltimetabling/TimeTableApp.java

Reply:
Updated.

Reply:
LGTM